crates/ra_hir_def/src/body/lower.rs

```diff
@@ -679,10 +660,20 @@ impl ExprCollector<'_> {
                     suffix: suffix.into_iter().map(|p| self.collect_pat(p)).collect(),
                 }
             }
+            ast::Pat::LiteralPat(lit) => {
+                if let Some(ast_lit) = lit.literal() {
+                    let expr = Expr::Literal(ast_lit.kind().into());
+                    let expr_ptr = Either::Left(AstPtr::new(&ast::Expr::Literal(ast_lit)));
+                    let src = Ok(InFile::new(self.expander.current_file_id, expr_ptr));
+                    let expr_id = self.make_expr(expr, src);
```

Comment:
I think you can just use `self.alloc_expr(expr, expr_ptr)` instead of constructing `src` yourself here?

Reply:
That is a bit nicer, thanks for the suggestion.